deployments/scripts/migration-test

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env bash
+# CI script to test migration compatibility. Orchestrates the following:
+#
+#   * [x] checks out prior tag and builds its binaries and tests
+#   * [x] creates local devnet via that target version
+#   * [x] make sure devnet has very fast voting
+#   * [x] runs smoke tests against devnet, to generate txs
+#   * [x] initialize smoke-test pcli wallet locally
+#   * [x] submit governance proposal for chain halt
+#   * [x] vote for proposal to pass
+#   * [x] wait for halt
+#   * [x] stops devnet ~ ends phase 1
+#   * [x] migrates devnet via latest (i.e. HEAD) pd version ~ begins phase 2
+#   * [x] restarts devnet
+#   * [ ] re-runs smoke tests, from latest version, to validate
```

Comment:
perhaps we can pick a subset of tests, so we don't end up running the entire smoke test suite twice. _or_, perhaps we should do it the other way around? we know previous releases work, but we're most interested in tests passing after the migration.